### PR TITLE
Ensure that exception is not swallowed by finally block

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Some cases the exception was not caught but swallowed by finally block.
Ensure we rethrow the exception after writing error.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by alerickson — APPROVED** [review]

File: src/code/InstallHelper.cs [hunk 1]
```diff
@@ -683,6 +683,16 @@ private List<PSResourceInfo> FilterByInstalledPkgs(List<PSResourceInfo> packages
                         }
                     }
                 }
+                catch (Exception e)
```
Comment (anamnavi) [inline src/code/InstallHelper.cs:686]: this method is not called anymore. we can delete it.

File: src/code/PublishPSResource.cs [hunk 2]
```diff
@@ -505,6 +505,16 @@ protected override void EndProcessing()
                     }
                 }
             }
+            catch (Exception e)
+            {
+                WriteError(new ErrorRecord(
```
Comment (anamnavi) [inline src/code/PublishPSResource.cs]: perhaps make this a ThrowTerminatingError()

File: src/code/UpdateModuleManifest.cs [hunk 3]
```diff
@@ -1020,6 +1031,16 @@ private void CreateModuleManifestForWinPSHelper(Hashtable parsedMetadata, string
                 WriteVerbose($"Moving '{newTmpModuleManifestPath}' to '{resolvedManifestPath}'");
                 Utils.MoveFiles(newTmpModuleManifestPath, resolvedManifestPath, overwrite: true);
             }
+            catch (Exception e)
```
Comment (anamnavi) [inline src/code/UpdateModuleManifest.cs:1031]: the method above is not catching this thrown error. Should we add a try/catch above?

File: src/code/UpdateModuleManifest.cs [hunk 4]
```diff
@@ -668,6 +668,17 @@ private void CreateModuleManifestHelper(Hashtable parsedMetadata, string resolve
                 WriteVerbose($"Moving '{tmpModuleManifestPath}' to '{resolvedManifestPath}'");
                 Utils.MoveFiles(tmpModuleManifestPath, resolvedManifestPath, overwrite: true);
             }
+            catch (Exception e)
```
Comment (anamnavi) [inline src/code/UpdateModuleManifest.cs:671]: the method above is not catching this thrown error. Should we add a try/catch above?

**Review by anamnavi — CHANGES_REQUESTED** [review]

File: src/code/PublishPSResource.cs (hunk 2 above)
Comment (adityapatwardhan) [reply src/code/PublishPSResource.cs]: Fixed

File: src/code/UpdateModuleManifest.cs (hunk 4 above)
Comment (adityapatwardhan) [reply src/code/UpdateModuleManifest.cs:671]: Removing the rethrow. The above layer checks for ErrorRecord and throws terminating error.

File: src/code/InstallHelper.cs (hunk 1 above)
Comment (adityapatwardhan) [reply src/code/InstallHelper.cs:686]: This method seems to be used. I will not remove it for now.

File: src/code/UpdateModuleManifest.cs (hunk 3 above)
Comment (adityapatwardhan) [reply src/code/UpdateModuleManifest.cs:1031]: sane as above, removed the rethrow

**Review by anamnavi — APPROVED** [review]
LGTM

**Pull request merged**